com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/StagingResolver.java

```diff
@@ -0,0 +1,25 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.StartingInfo;
+import org.cloudfoundry.client.lib.domain.PackageState;
+
+import com.sap.cloud.lm.sl.cf.process.steps.ExecutionWrapper;
+import com.sap.cloud.lm.sl.common.util.Pair;
+
+public abstract class StagingResolver {

```

Comment:
You don't need this abstract class. Move the logic to CFStagingResolver, rename it to StagingResolver and extend it for XSA.